build.gradle

```diff
@@ -14,6 +15,18 @@ repositories {
   jcenter()
 }
 
+idea {
+  module {
+    downloadJavadoc = false
+    downloadSources = false
+  }
+}
+
+tasks.withType(AbstractCompile).all {
+  sourceCompatibility = JavaVersion.VERSION_1_8
```

Comment:
Generally I would lean towards setting this as `sourceCompatibility = '1.8'` at the project level rather than on each task.

Reply:
I considered the comment. I looked at source codes (the http://bit.ly/2hnsFkX and http://bit.ly/2hoI9VP) and I agree with the comment - it is possible to register compatibility at the level of the project.